x-pack/plugins/enterprise_search/server/lib/connectors/add_connector.ts

```diff
@@ -23,11 +27,12 @@ const createConnector = async (
   document: ConnectorDocument,
   client: IScopedClusterClient,
   language: string | null,
-  deleteExisting: boolean
+  deleteExisting: boolean,
+  createDocumentIndex: boolean = true
```

Comment:
note: Doesn't necessarily need to be addressed here, but this is starting to look like too many inputs to this function and too many branches. Probably time to start splitting it up.

Reply:
Yeah, part of me felt that we probably need one function/builder for constructing the ConnectorDocument, and a separate function to index it. I should also have taken a closer look at @navarone-feekery's pr: https://github.com/elastic/kibana/pull/139651. Nav, let's get yours in first, and then I'll work on a refactor.

Reply:
@seanstory sorry to have blocked you! that PR is merged now 🙂

Reply:
:) no worries Nav, you didn't block me! It'll be good that I can build off your work.